src/twisted/newsfragments/11856.bugfix

```diff
@@ -0,0 +1 @@
+Python 3.12 removed assertRaisesRegexp unittest TestCase method alias. Replacing it with assertRaisesRegex fixes the failing test.
```

Comment:
I am not sure about this description.

Is this a bug?

Which tests are failing?

I can see that trunk is green.

![image](https://user-images.githubusercontent.com/204609/236626140-40e23bb5-de78-4847-b97a-98d0936c55eb.png)

---------

I guess that this is a change for a future 3.12 release or for a current 3.12 alpha release.

The  release notes are for users of Twisted.
This looks like some internal cleanup of  our tests.

so this can can be left empty with just an 11856.misc to confirm that we have considered the release notes for this PR, but ended with nothing of interst for the "Twisted audience"

Reply:
I skimmed through the development process doc and missed the `misc` option, that's indeed more suitable. I'll adjust the PR accordingly.